src/main/java/org/jenkinsci/deprecatedusage/DeprecatedUsage.java

```diff
@@ -125,7 +107,42 @@ void methodCalled(String className, String name, String desc) {
                     }
                 }
             }
+    }
+
+    /**
+     * Returns true if given class should be analyzed
+     *
+     * @see Options
+     */
+    private boolean shouldAnalyze(String className)  {
+
+        if (className.endsWith("DefaultTypeTransformation")) {
+            // various DefaultTypeTransformation#box signatures seem false positive in plugins written in Groovy
+            return false;
+        }
+
+        // if an additionalClasses file is specified, and this matches, we ignore Options' includeJavaCoreClasses or onlyIncludeJenkinsClasses
+        // values, given the least surprise is most likely that if the user explicitly passed a file, s/he does want it to be analyzed
+        // even if coming from java.*, javax.*, or not from Jenkins core classes itself
+        if (Options.get().additionalClassesFile != null &&
+                Options.getAdditionalClasses().stream().anyMatch(className::startsWith)) {
+            return true;
+        }
+
+        if(Options.get().onlyAdditionalClasses) {
+            return false;
         }
+
+        // Calls to java and javax are ignored by default if not explicitly requested
+        if(className.startsWith("java/") || className.startsWith("javax")) {
```

Comment:
Isn't it `isJavaClass`?

Reply:
Good catch, addressed in https://github.com/jenkins-infra/deprecated-usage-in-plugins/pull/11/commits/13e0045a35ae74f0bb380ec7280a7824026189d1